src/scormAPI2004.js

```diff
@@ -915,4 +915,34 @@
   function adlNavRequestValidJump(_target) {
     return "unknown";
   }
+
+  /**
+   * Determine the completion_status of a cmi object
+   */
+  function getCompletionStatus(cmi) {
+    var completion_status = cmi._completion_status;
+
+    if (cmi.completion_threshold && cmi.progress_measure) {
```

Comment:
I would suggest using the private variables that start with an underscore, to avoid going through the getter methods. The SCORM standard says that the getter method should throw a `Data Model Element Value Not Initialized` error when certain attributes are not set, so if we use the getters in other methods, those errors would be thrown.

Using the private variables bypasses the getter methods, and therefore avoids the errors. To be fair, these errors do not exist yet, but I'd prefer to avoid future problems.
```suggestion
    if (cmi._completion_threshold && cmi._progress_measure) {
```